### PR TITLE
Introduce rosaControlPlane for managed kubernetes

### PR DESCRIPTION
**Pull request by enxebre**
**What type of PR is this?**

**What this PR does / why we need it**:

This introduces support [ROSA](https://aws.amazon.com/rosa/) analogous to the EKS implementation.

This branch introduces a new ROSAControlPlane CRD and controller which enables functional ROSA Cluster creation with the proper yaml definition. We plan to iterate and follow up to flesh out design and implementation for the API, cluster deletion, day 2 changes, machine management via rosaMachinePools, etc.
This first iteration is meant to be little more than a starting common ground for collaboration.

Let's discuss on next community meeting next steps.

cc @vincepri @richardcase

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by enxebre** [comment]
Planning to bring this up in next community call Mon 3 Jul

**Comment by enxebre** [comment]
>Planning to bring this up in next community call Mon 3 Jul

There's no meeting today, so 10Jul.

**Comment by vincepri** [comment]
@enxebre Can you rebase?

**Review by Ankitasw — COMMENTED** [review]
Shall this be an experimental feature first? 
If yes, then it would require a bit of refactoring, as we follow for exp APIs and controllers.

**Comment by richardcase** [comment]
> Shall this be an experimental feature first and then be graduated? If yes, then it would require a bit of refactoring, as we follow for exp APIs and controllers.

Good point @Ankitasw . Yes i think we do need to sit this behind a feature flag initially like we did when we added other significant new features.

**Comment by muraee** [comment]
it would be good if we can add an OWNER file or something to allow us to approve PRs related to the ROSA provider without needing to bother the maintainers every time. 
cc @richardcase @Ankitasw thoughts?

File: controllers/rosacluster_controller.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+
+// ROSAClusterReconciler reconciles ROSACluster.
+type ROSAClusterReconciler struct {
+	client.Client
+	Recorder         record.EventRecorder
+	WatchFilterValue string
+}
+
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=ROSAclusters,verbs=get;list;watch;update;patch;delete
+// +kubebuilder:rbac:groups=infrastructure.cluster.x-k8s.io,resources=ROSAclusters/status,verbs=get;update;patch
```
Comment (prometherion) [inline controllers/rosacluster_controller.go:53]: Shouldn't be these resources in lowercase?

**Comment by vincepri** [comment]
> it would be good if we can add an OWNER file or something to allow us to approve PRs related to the ROSA provider without needing to bother the maintainers every time. cc @richardcase @Ankitasw thoughts?

Adding a sub-owner file sounds good to me, although I'd generally prefer a steady stream of contribution/reviews before we add approver rights.

**Pull request closed without merging**